helix-view/src/document.rs

```diff
@@ -603,6 +603,13 @@ impl Document {
         self.set_language(language_config, Some(config_loader));
     }
 
+    /// Set the programming language for the file if you know the language but don't have the
+    /// [`syntax::LanguageConfiguration`] for it.
+    pub fn set_language3(&mut self, language_id: &str, config_loader: Arc<syntax::Loader>) {
```

Comment:
Let's call this `set_language_by_language_id`?

Reply:
done.